### PR TITLE
Add singletonPage

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: src/Database/LSMTree/Internal/PageAcc1.hs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+-- | Create a singleton page, also returning the overflow value bytes.
+singletonPage
+    :: SerialisedKey
+    -> Entry SerialisedValue BlobSpan
+    -> (RawPage, RawBytes)
+singletonPage k (Insert v) = runST $ do
+    -- allocate bytearray
+    ba <- P.newByteArray pageSize :: ST s (P.MutableByteArray s)
+    P.fillByteArray ba 0 pageSize 0
```
Comment (mheinzel) [inline src/Database/LSMTree/Internal/PageAcc1.hs:30]: Makes the code more obviously correct, good. You could leave a TODO to try optimising it away later (during benchmarking & optimisation phase), if you think it makes sense.

**Review by mheinzel — APPROVED** [review]
There's the thing about handling `Mupdate`, which I agree would be good to address now that you're working on this code already. Apart from that, everything looks very sensible to me.

File: src/Database/LSMTree/Internal/PageAcc1.hs (hunk 1 above)
Comment (phadej) [reply src/Database/LSMTree/Internal/PageAcc1.hs:30]: I don't think it makes. `memset` is cheap.

**Comment by phadej** [comment]
Ah, good point. `Mupdate` has value too, I forgot about that. I'll add that too.

**Review by dcoutts — APPROVED** [review]
LGTM

**Pull request merged**